src/win/tty.c

```diff
@@ -463,25 +492,12 @@ static void uv_tty_queue_read_line(uv_loop_t* loop, uv_tty_t* handle) {
   }
   assert(handle->tty.rd.read_line_buffer.base != NULL);
 
-  /* Duplicate the console handle, so if we want to cancel the read, we can */
-  /* just close this handle duplicate. */
-  if (handle->tty.rd.read_line_handle == NULL) {
-    HANDLE this_process = GetCurrentProcess();
-    r = DuplicateHandle(this_process,
-                        handle->handle,
-                        this_process,
-                        &handle->tty.rd.read_line_handle,
-                        0,
-                        0,
-                        DUPLICATE_SAME_ACCESS);
-    if (!r) {
-      handle->tty.rd.read_line_handle = NULL;
-      SET_REQ_ERROR(req, GetLastError());
-      uv_insert_pending_req(loop, (uv_req_t*)req);
-      goto out;
-    }
-  }
-
+  /* Reset flags  No locking is required since there cannot be a line read
+     in progress. QueueUserWorkItem probably acts as a memory barrier,
+     but we add an explicit one to be extra safe */
+  uv__read_console_started = FALSE;
+  uv__read_console_trap = FALSE;
+  MemoryBarrier();
```

Comment:
Agreed. Everyone who reviewed the change before asked why there was no locking here, so I thought the explicit MemoryBarrier() would make it more readable. But the comment should be enough. I'll remove it.